### DISCUSSION TITLE
Set NODE_ENV

### DISCUSSION THREAD

**Discussion by carlocorradini** [TypeStrong/ts-node#1281] [Q&A] [answered]
How can i set NODE_ENV without using any global environment variable or using third party packages like cross-env?

ts-node does not set any value to NODE_ENV so it's undefined.
In various part of the code I have a check if NODE_ENV is production or undefined that set all configuration for 'production ready'. This is a problem for me :(.
Currently to run ts-node in development I have made this simple script that uses cross-env to inject the NODE_ENV environment variable:
```typescript
npx cross-env NODE_ENV=development ts-node index.ts
```

I think it can be a nice idea to have an option that set NODE_ENV variable according to our needs.

For example:
1. programmatic option:
```bash
$ npx ts-node --node-env development 
```
2. tsconfig.json
```json
{
  "ts-node": {
      "node-env": "development"
  },
  "compilerOptions": {}
}
```

Thanks in advance! 🎉

**Comment by cspotcode** [comment, marked as answer]
This is far out of scope for `ts-node`.  Any tool or library which accomplishes this for `node` will also work for `ts-node`, so I recommend using one of those.
